web-ui/github.ml

```diff
@@ -337,24 +337,19 @@ let repo_handle ~meth ~owner ~name ~repo path =
       | Aborted | Failed _ | Passed | Undefined _ -> Lwt.return None
     in
     let cancel_many (commit: Client.Commit.t) (job_infos : Client.job_info list) =
-      let open Lwt.Infix in
-      Lwt_list.filter_map_p (fun job_info -> can_cancel commit job_info) job_infos >>= fun l ->
-        Lwt_list.map_p (fun (ji, j) -> Current_rpc.Job.cancel j |> Lwt_result.map @@ fun () -> ji) l >>= fun x ->
-          let success = ref [] in
-          let failed = ref 0 in
-          let log_error r =
-          match r with
-          | Ok(ji) -> success := ji :: !success
-          | Error (`Capnp ex) ->
-            incr failed;
-            Log.err (fun f -> f "Error cancelling job: %a" Capnp_rpc.Error.pp ex)
-          in
-          let _ = List.map log_error x in
-          Lwt.return (!success, !failed)
+      Lwt_list.fold_left_s (fun (success, failed) job_info ->
+        can_cancel commit job_info >>= function
```

Comment:
@novemberkilo I’m wondering if it might be more better to just call `Current_rpc.Job.cancel`. In the worse case scenario (not really, but see bellow) it does not do anything, whereas in the current state two rpc calls are done (one to see if the job is cancellable, and the other to cancel it).

In practice: https://github.com/ocurrent/ocurrent/blob/e0f32037e1fff945b53fbba13489aa8d250115ad/lib/job.ml#L92, shows that the "wrong cancellation" is logged. It might be nice to have a separate function that actually ignores when jobs are already done. Such function could be added to ocurrent as `Current_rpc.Job.cancel_if_cancellable` or something of the sort.

Reply:
Thanks for the suggestion - I'll try this out on `ocaml-ci` as I bring your refactoring across to it.